src/objects/zcl_abapgit_object_vcls.clas.abap

```diff
@@ -141,13 +171,52 @@ CLASS ZCL_ABAPGIT_OBJECT_VCLS IMPLEMENTATION.
 
 
   METHOD zif_abapgit_object~is_active.
-    rv_active = is_active( ).
+
+    DATA lv_changedate TYPE vcldir-changedate.
+
+    SELECT SINGLE changedate INTO lv_changedate FROM vcldir
+      WHERE vclname = ms_item-obj_name.
+
+    IF lv_changedate IS NOT INITIAL.
+* see logic in function module VIEWCLUSTER_GET_DEFINITION
+      rv_active = abap_true.
+    ELSE.
+      rv_active = abap_false.
+    ENDIF.
+
   ENDMETHOD.
 
 
   METHOD zif_abapgit_object~is_locked.
 
-    rv_is_locked = abap_false.
+
+    DATA:
+      ls_sellist TYPE vimsellist,
+      lt_sellist TYPE scprvimsellist.
+
+    ls_sellist-viewfield  = 'VCLNAME'.
+    ls_sellist-operator   = 'EQ'.
+    ls_sellist-value      = me->ms_item-obj_name.
+    APPEND ls_sellist TO lt_sellist.
+
+    rv_is_locked = check_lock( iv_viewname = 'V_VCLDIR'
+                               it_sellist  = lt_sellist ).
+
+    IF rv_is_locked = abap_false.
+      rv_is_locked = check_lock( iv_viewname = 'V_VCLSTRUC'
+                                 it_sellist  = lt_sellist ).
+      IF rv_is_locked = abap_false.
+
+        rv_is_locked = check_lock( iv_viewname = 'V_VCLSTDEP'
+                                   it_sellist  = lt_sellist ).
+
+        IF rv_is_locked = abap_false.
+          rv_is_locked = check_lock( iv_viewname = 'V_VCLMF'
+                                     it_sellist  = lt_sellist ).
+
+        ENDIF.
+      ENDIF.
+    ENDIF.
 
```

Comment:
Can you reduce the nesting of the conditionals?

Reply:
Will change when I reworke the lock method